server/src-test/Main.hs

```diff
@@ -31,7 +31,10 @@ import qualified Hasura.Server.MigrateSpec    as MigrateSpec
 
 data TestSuites
   = AllSuites !RawConnInfo
-  | SingleSuite !TestSuite
+  -- ^ Run all test suites. It probably doesn't make sense to be able to specify additional 
+  -- hspec args here.
```

Comment:
I’m not totally convinced by this comment! I think it makes as much sense to specify additional hspec arguments  when running all the test suites as it does when running just one.

Originally I considered making these truly different test suites, in the Cabal sense. We could have `test-suite graphql-engine-unit-tests` and `test-suite graphql-engine-postgres-tests`. However, working with multiple test suites can be a little awkward, especially with `stack`, so I decided to only have one test suite that can be configured via options. `cabal new-run` makes this a lot simpler, though, so maybe that would be a simpler solution now. I don’t feel strongly either way.

Reply:
Does `--match` fail when nothing matches? If not I suppose match and skip would both be fine to pass to all.

Reply:
I'm fine if the behavior changes going forward if someone decides it makes sense (hopefully the comment reflects that)

Reply:
To your original question about `--match`: I don’t know what it does if nothing matches. I’d expect it to pass (and run nothing), but I haven’t checked.

I think that if we want to expose `hspec` options on the command line (which is reasonable), we should not treat `TestSuites` like subcommands, but more like other flags like `--match` and `--skip`. So if you wanted, you could pass `--suite=unit`, and it would select those tests.

Reply:
I'm fine with any of those changes you're suggesting, but don't have much of an opinion. Would you agree that can happen later? I'd love to get these fixes merged if possible.